bot/cogs/moderation/silence.py

```diff
@@ -11,9 +11,14 @@
 from bot.constants import Channels, Emojis, Guild, MODERATION_ROLES, Roles
 from bot.converters import HushDurationConverter
 from bot.utils.checks import with_role_check
+from bot.utils.scheduling import Scheduler
 
 log = logging.getLogger(__name__)
 
+SilencedChannel = NamedTuple(
+    "SilencedChannel", [("ctx", Context), ("delay", int)]
+)
+
```

Comment:
This should be a class definition https://github.com/python-discord/bot/blob/dd11ba4e6d6c33005c4eab10f4b8b2882f6f1f3d/bot/cogs/help_channels.py#L60-L64 as an example

Reply:
Why should it be a class? Isn't it just doing the same thing in a longer approach?

Reply:
It's clearer syntax, and there's no reason to avoid a class definition here

Reply:
Alright, I can do that, but I've noticed you replaced `ctx` with `callback`, should I just pass `lambda: ctx.invoke(self.unsilence)` or should I only go with `ctx.invoke` and passing the `self.unsilence` from `_scheduled_task` or did you have something else in mind?

Reply:
Pass in `ctx.invoke(self.unsilence)` without awaiting it. `ctx.invoke` is a coroutine function, which means it returns a coroutine object. A coroutine object is awaitable, hence satisfying the annotation `t.Awaitable`. Perhaps `coroutine` is a better name than `callback` but I think it's clear enough either way.

Reply:
Yes but that will produce RuntimeWarning if I unsilence prematurely since it won't be awaited

Reply:
The above was just an example from a different module, not as a direct suggestion.
I think the names are fine with how they are now if you don't want to change the structure, so just need to adjust the syntax and add a docstring

Reply:
This is what I did to solve that issue https://github.com/python-discord/bot/blob/master/bot/cogs/help_channels.py#L843

Reply:
Oh, that's clever, so should I go with that, or as @Numerlor said, keep it as it was and only turn it into a class?

Reply:
Sorry, I forgot what I was looking at for a second 😅. You don't need to pass a coroutine. This would only really be beneficial if you needed to support different coroutines, as was the case in the help channel cog.